lib/apiservers/engine/backends/volume.go

```diff
@@ -15,32 +15,79 @@
 package vicbackends
 
 import (
+	"encoding/json"
 	"fmt"
 	"net/http"
+	"os"
 
 	derr "github.com/docker/docker/errors"
 
 	"github.com/docker/engine-api/types"
 	"github.com/vmware/vic/lib/apiservers/portlayer/client/storage"
+	"github.com/vmware/vic/lib/apiservers/portlayer/models"
+	"github.com/vmware/vic/lib/guest"
 	"github.com/vmware/vic/pkg/trace"
 )
 
+//Volume : struct which defines the docker personalities view of a Volume
 type Volume struct {
 	ProductName string
 }
 
+type volumeMetadata struct {
+	Driver string
+	Name   string
+	Labels map[string]string
+}
+
+//Volumes : docker personality implementation for VIC
 func (v *Volume) Volumes(filter string) ([]*types.Volume, []string, error) {
 	return nil, make([]string, 0), fmt.Errorf("%s does not implement volume.Volumes", v.ProductName)
```

Comment:
Can you not return `nil` in place of this slice? Or will the client crash or something? Just curious.